v7.4 Email NPE when Alerting a Streaming Analytic

```
java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:233)
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:180)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at stroom.analytics.impl.RuleEmailTemplatingService.lambda$buildContext$12(RuleEmailTemplatingService.java:125)
	at stroom.util.NullSafe.consume(NullSafe.java:701)
	at stroom.analytics.impl.RuleEmailTemplatingService.buildContext(RuleEmailTemplatingService.java:123)
	at stroom.analytics.impl.RuleEmailTemplatingService.renderAlertEmail(RuleEmailTemplatingService.java:32)
	at stroom.analytics.impl.EmailSender.send(EmailSender.java:90)
	at stroom.analytics.impl.DetectionConsumerFactory$3.accept(DetectionConsumerFactory.java:136)
	at stroom.analytics.impl.DetectionConsumerProxy.writeRecord(DetectionConsumerProxy.java:238)
	at stroom.analytics.impl.DetectionConsumerProxy.accept(DetectionConsumerProxy.java:120)
	at stroom.analytics.impl.AbstractAnalyticFieldListConsumer.acceptFieldValues(AbstractAnalyticFieldListConsumer.java:57)
	at stroom.analytics.impl.StreamingAnalyticFieldListConsumer.acceptFieldValues(StreamingAnalyticFieldListConsumer.java:13)
	at stroom.search.extraction.FieldListConsumerHolder.acceptFieldValues(FieldListConsumerHolder.java:23)
	at stroom.search.extraction.DynamicSearchResultOutputFilter.processFields(DynamicSearchResultOutputFilter.java:56)
	at stroom.search.extraction.AbstractFieldFilter.endElement(AbstractFieldFilter.java:96)
	at net.sf.saxon.event.ContentHandlerProxy.endElement(ContentHandlerProxy.java:412)
	at net.sf.saxon.event.ProxyReceiver.endElement(ProxyReceiver.java:185)
	at net.sf.saxon.event.SequenceNormalizer.endElement(SequenceNormalizer.java:230)
	at net.sf.saxon.event.NamespaceReducer.endElement(NamespaceReducer.java:235)
	at net.sf.saxon.event.ComplexContentOutputter.endElement(ComplexContentOutputter.java:559)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:356)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:300)
	at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
	at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:300)
	at net.sf.saxon.expr.instruct.ApplyTemplates.processLeavingTail(ApplyTemplates.java:255)
	at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
	at net.sf.saxon.expr.instruct.Instruction.process(Instruction.java:132)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:353)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:300)
	at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
	at net.sf.saxon.trans.rules.TextOnlyCopyRuleSet.process(TextOnlyCopyRuleSet.java:71)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:495)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:746)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at net.sf.saxon.event.ContentHandlerProxy.close(ContentHandlerProxy.java:295)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.SequenceNormalizer.close(SequenceNormalizer.java:248)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.ComplexContentOutputter.close(ComplexContentOutputter.java:621)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:753)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at net.sf.saxon.event.ContentHandlerProxy.close(ContentHandlerProxy.java:295)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.SequenceNormalizer.close(SequenceNormalizer.java:248)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.ComplexContentOutputter.close(ComplexContentOutputter.java:621)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:753)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at stroom.pipeline.filter.AbstractXMLFilter.endDocument(AbstractXMLFilter.java:158)
	at stroom.pipeline.filter.AbstractXMLFilter.endDocument(AbstractXMLFilter.java:158)
	at stroom.pipeline.filter.SplitFilter.fireRemainingEvents(SplitFilter.java:440)
	at stroom.pipeline.filter.SplitFilter.endDocument(SplitFilter.java:199)
	at org.apache.xerces.parsers.AbstractSAXParser.endDocument(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl.endEntity(Unknown Source)
	at org.apache.xerces.impl.XMLEntityManager.endEntity(Unknown Source)
	at org.apache.xerces.impl.XMLEntityScanner.load(Unknown Source)
	at org.apache.xerces.impl.XML11EntityScanner.skipSpaces(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl$TrailingMiscDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at stroom.pipeline.parser.AbstractParser.parse(AbstractParser.java:185)
	at stroom.pipeline.parser.AbstractParser.lambda$createProcessors$0(AbstractParser.java:86)
	at stroom.pipeline.factory.PipelineImpl.process(PipelineImpl.java:129)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.processNestedStreams(AbstractProcessorTaskExecutor.java:459)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.process(AbstractProcessorTaskExecutor.java:340)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.exec(AbstractProcessorTaskExecutor.java:239)
	at stroom.processor.impl.DataProcessorTaskHandler.exec(DataProcessorTaskHandler.java:164)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$0(DataProcessorTaskHandler.java:92)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$2(TaskContextFactoryImpl.java:253)
	at stroom.util.logging.LocationAwareLambdaLogger.logDurationIfDebugEnabled(LocationAwareLambdaLogger.java:307)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$4(TaskContextFactoryImpl.java:253)
	at stroom.util.pipeline.scope.PipelineScopeRunnable.scopeResult(PipelineScopeRunnable.java:39)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$5(TaskContextFactoryImpl.java:250)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$6(TaskContextFactoryImpl.java:262)
	at stroom.security.impl.SecurityContextImpl.useAsReadResult(SecurityContextImpl.java:406)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
	at jdk.proxy2/jdk.proxy2.$Proxy122.useAsReadResult(Unknown Source)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$7(TaskContextFactoryImpl.java:262)
	at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:338)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
	at jdk.proxy2/jdk.proxy2.$Proxy122.asUserResult(Unknown Source)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$11(TaskContextFactoryImpl.java:260)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$1(DataProcessorTaskHandler.java:92)
	at stroom.security.impl.SecurityContextImpl.useAsReadResult(SecurityContextImpl.java:406)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$2(DataProcessorTaskHandler.java:87)
	at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:338)
	at stroom.processor.impl.DataProcessorTaskHandler.exec(DataProcessorTaskHandler.java:87)
	at stroom.processor.impl.DataProcessorTaskFactory$RunnableFactoryImpl.lambda$create$0(DataProcessorTaskFactory.java:166)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)




Error sending alert email Email{
	id=&lt;1072601236.0.1726747063489@server&gt;
	sentDate=null
	fromRecipient=Recipient{name='Stroomy Analytics', address='someone@somewhere', type=null},
	replyToRecipients=[Recipient{name='Stroomy Analytics', address='someone@somewhere', type=null}],
	bounceToRecipient=null,
	text='null',
	textHTML='null',
	textCalendar='null (method: null)',
	contentTransferEncoding='quoted-printable',
	subject='null',
	recipients=[Recipient{name='someone@somewhere', address='someone@somewhere', type=To}]
} using user (xxxx) at server:nnnn - Failed to send email [ID: '&lt;1072601236.0.1726747063489@server&gt;'], reason: Third party error 
org.simplejavamail.mailer.internal.MailerException: Failed to send email [ID: '&lt;1072601236.0.1726747063489@server&gt;'], reason: Third party error
	at org.simplejavamail.mailer.internal.SendMailClosure.handleException(SendMailClosure.java:85)
	at org.simplejavamail.mailer.internal.SendMailClosure.executeClosure(SendMailClosure.java:72)
	at org.simplejavamail.mailer.internal.AbstractProxyServerSyncingClosure.run(AbstractProxyServerSyncingClosure.java:56)
	at org.simplejavamail.mailer.internal.MailerImpl.sendMail(MailerImpl.java:364)
	at org.simplejavamail.mailer.internal.MailerImpl.sendMail(MailerImpl.java:349)
	at stroom.analytics.impl.EmailSender.send(EmailSender.java:137)
	at stroom.analytics.impl.DetectionConsumerFactory$3.accept(DetectionConsumerFactory.java:136)
	at stroom.analytics.impl.DetectionConsumerProxy.writeRecord(DetectionConsumerProxy.java:238)
	at stroom.analytics.impl.DetectionConsumerProxy.accept(DetectionConsumerProxy.java:120)
	at stroom.analytics.impl.AbstractAnalyticFieldListConsumer.acceptFieldValues(AbstractAnalyticFieldListConsumer.java:57)
	at stroom.analytics.impl.StreamingAnalyticFieldListConsumer.acceptFieldValues(StreamingAnalyticFieldListConsumer.java:13)
	at stroom.search.extraction.FieldListConsumerHolder.acceptFieldValues(FieldListConsumerHolder.java:23)
	at stroom.search.extraction.DynamicSearchResultOutputFilter.processFields(DynamicSearchResultOutputFilter.java:56)
	at stroom.search.extraction.AbstractFieldFilter.endElement(AbstractFieldFilter.java:96)
	at net.sf.saxon.event.ContentHandlerProxy.endElement(ContentHandlerProxy.java:412)
	at net.sf.saxon.event.ProxyReceiver.endElement(ProxyReceiver.java:185)
	at net.sf.saxon.event.SequenceNormalizer.endElement(SequenceNormalizer.java:230)
	at net.sf.saxon.event.NamespaceReducer.endElement(NamespaceReducer.java:235)
	at net.sf.saxon.event.ComplexContentOutputter.endElement(ComplexContentOutputter.java:559)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:356)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:300)
	at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
	at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:300)
	at net.sf.saxon.expr.instruct.ApplyTemplates.processLeavingTail(ApplyTemplates.java:255)
	at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
	at net.sf.saxon.expr.instruct.Instruction.process(Instruction.java:132)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:353)
	at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:300)
	at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
	at net.sf.saxon.trans.rules.TextOnlyCopyRuleSet.process(TextOnlyCopyRuleSet.java:71)
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:495)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:746)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at net.sf.saxon.event.ContentHandlerProxy.close(ContentHandlerProxy.java:295)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.SequenceNormalizer.close(SequenceNormalizer.java:248)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.ComplexContentOutputter.close(ComplexContentOutputter.java:621)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:753)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at net.sf.saxon.event.ContentHandlerProxy.close(ContentHandlerProxy.java:295)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.SequenceNormalizer.close(SequenceNormalizer.java:248)
	at net.sf.saxon.event.ProxyReceiver.close(ProxyReceiver.java:102)
	at net.sf.saxon.event.ComplexContentOutputter.close(ComplexContentOutputter.java:621)
	at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:753)
	at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
	at net.sf.saxon.jaxp.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:173)
	at stroom.pipeline.filter.XsltFilter.endDocument(XsltFilter.java:305)
	at stroom.pipeline.filter.AbstractXMLFilter.endDocument(AbstractXMLFilter.java:158)
	at stroom.pipeline.filter.AbstractXMLFilter.endDocument(AbstractXMLFilter.java:158)
	at stroom.pipeline.filter.SplitFilter.fireRemainingEvents(SplitFilter.java:440)
	at stroom.pipeline.filter.SplitFilter.endDocument(SplitFilter.java:199)
	at org.apache.xerces.parsers.AbstractSAXParser.endDocument(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl.endEntity(Unknown Source)
	at org.apache.xerces.impl.XMLEntityManager.endEntity(Unknown Source)
	at org.apache.xerces.impl.XMLEntityScanner.load(Unknown Source)
	at org.apache.xerces.impl.XML11EntityScanner.skipSpaces(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl$TrailingMiscDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at stroom.pipeline.parser.AbstractParser.parse(AbstractParser.java:185)
	at stroom.pipeline.parser.AbstractParser.lambda$createProcessors$0(AbstractParser.java:86)
	at stroom.pipeline.factory.PipelineImpl.process(PipelineImpl.java:129)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.processNestedStreams(AbstractProcessorTaskExecutor.java:459)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.process(AbstractProcessorTaskExecutor.java:340)
	at stroom.core.dataprocess.AbstractProcessorTaskExecutor.exec(AbstractProcessorTaskExecutor.java:239)
	at stroom.processor.impl.DataProcessorTaskHandler.exec(DataProcessorTaskHandler.java:164)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$0(DataProcessorTaskHandler.java:92)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$2(TaskContextFactoryImpl.java:253)
	at stroom.util.logging.LocationAwareLambdaLogger.logDurationIfDebugEnabled(LocationAwareLambdaLogger.java:307)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$4(TaskContextFactoryImpl.java:253)
	at stroom.util.pipeline.scope.PipelineScopeRunnable.scopeResult(PipelineScopeRunnable.java:39)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$5(TaskContextFactoryImpl.java:250)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$6(TaskContextFactoryImpl.java:262)
	at stroom.security.impl.SecurityContextImpl.useAsReadResult(SecurityContextImpl.java:406)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
	at jdk.proxy2/jdk.proxy2.$Proxy122.useAsReadResult(Unknown Source)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$7(TaskContextFactoryImpl.java:262)
	at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:338)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
	at jdk.proxy2/jdk.proxy2.$Proxy122.asUserResult(Unknown Source)
	at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$11(TaskContextFactoryImpl.java:260)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$1(DataProcessorTaskHandler.java:92)
	at stroom.security.impl.SecurityContextImpl.useAsReadResult(SecurityContextImpl.java:406)
	at stroom.processor.impl.DataProcessorTaskHandler.lambda$exec$2(DataProcessorTaskHandler.java:87)
	at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:338)
	at stroom.processor.impl.DataProcessorTaskHandler.exec(DataProcessorTaskHandler.java:87)
	at stroom.processor.impl.DataProcessorTaskFactory$RunnableFactoryImpl.lambda$create$0(DataProcessorTaskFactory.java:166)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: jakarta.mail.MessagingException: No MimeMessage content
	at jakarta.mail.internet.MimeMessage.getContentStream(MimeMessage.java:1413)
	at jakarta.mail.internet.MimeBodyPart$MimePartDataHandler.getContentStream(MimeBodyPart.java:1717)
	at jakarta.mail.internet.MimeBodyPart.writeTo(MimeBodyPart.java:1672)
	at jakarta.mail.internet.MimeMessage.writeTo(MimeMessage.java:1911)
	at org.eclipse.angus.mail.smtp.SMTPTransport.sendMessage(SMTPTransport.java:1305)
	at org.simplejavamail.mailer.internal.util.TransportRunner.lambda$sendMessage$0(TransportRunner.java:61)
	at org.simplejavamail.mailer.internal.util.TransportRunner.sendUsingConnectionPool(TransportRunner.java:92)
	at org.simplejavamail.mailer.internal.util.TransportRunner.runOnSessionTransport(TransportRunner.java:77)
	at org.simplejavamail.mailer.internal.util.TransportRunner.sendMessage(TransportRunner.java:56)
	at org.simplejavamail.mailer.internal.SendMailClosure.executeClosure(SendMailClosure.java:69)
	... 111 common frames omitted
```

--- a/stroom-analytics/stroom-analytics-impl/src/main/java/stroom/analytics/impl/AnalyticsServiceImpl.java
+++ b/stroom-analytics/stroom-analytics-impl/src/main/java/stroom/analytics/impl/AnalyticsServiceImpl.java
@@ -50,6 +50,7 @@ public class AnalyticsServiceImpl implements AnalyticsService {
                 .addValue("name-1", "value-A")
                 .addValue("name-2", "value-B")
                 .addValue("name-3", "value-C")
+                .addValue("name-3", null)
                 .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 1L))
                 .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 2L))
                 .addLinkedEvents(new DetectionLinkedEvent(stroom, 2001L, 1L))

--- a/stroom-analytics/stroom-analytics-impl/src/main/java/stroom/analytics/impl/RuleEmailTemplatingService.java
+++ b/stroom-analytics/stroom-analytics-impl/src/main/java/stroom/analytics/impl/RuleEmailTemplatingService.java
@@ -121,8 +121,16 @@ public class RuleEmailTemplatingService {
         NullSafe.consume(detection.getDefunct(), val -> context.put("defunct", val));
 
         NullSafe.consume(detection.getValues(), values -> {
-            context.put("values", values.stream()
-                    .collect(Collectors.toMap(DetectionValue::getName, DetectionValue::getValue)));
+            // Collectors.toMap() doesn't like null values, shame
+            final Map<String, String> map = new HashMap<>(values.size());
+            values.stream()
+                    .filter(Objects::nonNull)
+                    .filter(detectionValue -> detectionValue.getName() != null)
+                    .forEach(detectionValue ->
+                            map.put(detectionValue.getName(), detectionValue.getValue()));
+            if (!map.isEmpty()) {
+                context.put("values", map);
+            }
         });
         NullSafe.consume(
                 detection.getLinkedEvents(),

--- a/stroom-analytics/stroom-analytics-impl/src/test/java/stroom/analytics/impl/TestRuleEmailTemplatingService.java
+++ b/stroom-analytics/stroom-analytics-impl/src/test/java/stroom/analytics/impl/TestRuleEmailTemplatingService.java
@@ -1,7 +1,9 @@
 package stroom.analytics.impl;
 
+import stroom.analytics.impl.Detection.Builder;
 import stroom.analytics.shared.EmailContent;
 import stroom.analytics.shared.NotificationEmailDestination;
+import stroom.ui.config.shared.AnalyticUiDefaultConfig;
 
 import org.junit.jupiter.api.Test;
 
@@ -12,7 +14,7 @@ class TestRuleEmailTemplatingService {
     @Test
     void renderAlertEmail() {
         final RuleEmailTemplatingService templatingService = new RuleEmailTemplatingService();
-        final Detection detection = getExampleDetection();
+        final Detection detection = getExampleDetection(true, true);
         NotificationEmailDestination emailDestination = NotificationEmailDestination.builder()
                 .subjectTemplate("{{ headline }}")
                 .bodyTemplate("{{ detectorVersion }} - {{ detectionRevision }}")
@@ -29,7 +31,7 @@ class TestRuleEmailTemplatingService {
     @Test
     void renderTemplate() {
         final RuleEmailTemplatingService templatingService = new RuleEmailTemplatingService();
-        final Detection detection = getExampleDetection();
+        final Detection detection = getExampleDetection(true, true);
         final String template = "{{ detectorVersion }} - {{ detectionRevision }}";
         final String output = templatingService.renderTemplate(detection, template);
 
@@ -37,13 +39,41 @@ class TestRuleEmailTemplatingService {
                 .isEqualTo(detection.getDetectorVersion() + " - " + detection.getDetectionRevision());
     }
 
-    private Detection getExampleDetection() {
+    @Test
+    void renderDefaultTemplate() {
+        final RuleEmailTemplatingService templatingService = new RuleEmailTemplatingService();
+        final Detection detection = getExampleDetection(true, true);
+        @SuppressWarnings("checkstyle:LineLength") final String template = new AnalyticUiDefaultConfig()
+                .getDefaultBodyTemplate();
+
+        final String output = templatingService.renderTemplate(detection, template);
+
+        assertThat(output)
+                .contains("name-4")
+                .contains(detection.getDetectorName());
+    }
+
+    @Test
+    void renderDefaultTemplate_noValues_noLinkedEvents() {
+        final RuleEmailTemplatingService templatingService = new RuleEmailTemplatingService();
+        final Detection detection = getExampleDetection(false, false);
+        @SuppressWarnings("checkstyle:LineLength") final String template = new AnalyticUiDefaultConfig()
+                .getDefaultBodyTemplate();
+
+        final String output = templatingService.renderTemplate(detection, template);
+        assertThat(output)
+                .doesNotContain("name-4")
+                .contains(detection.getDetectorName());
+    }
+
+    private Detection getExampleDetection(final boolean includeValues,
+                                          final boolean includeLinkedEvents) {
         final String stroom = "Test Environment";
         final String executionTime = "2024-02-29T17:48:40.396Z";
         final String detectTime = "2024-02-29T17:48:41.582Z";
         final String effectiveExecutionTime = "2024-02-29T16:00:00.000Z";
 
-        return Detection.builder()
+        final Builder builder = Detection.builder()
                 .withDetectTime(detectTime)
                 .withDetectorName("Example detector for test use.")
                 .withHeadline("The headline for the detection.")
@@ -56,14 +86,22 @@ class TestRuleEmailTemplatingService {
                 .withDefunct(false)
                 .withExecutionTime(executionTime)
                 .withExecutionSchedule("1hr")
-                .withEffectiveExecutionTime(effectiveExecutionTime)
-                .addValue("name-1", "value-A")
-                .addValue("name-2", "value-B")
-                .addValue("name-3", "value-C")
-                .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 1L))
-                .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 2L))
-                .addLinkedEvents(new DetectionLinkedEvent(stroom, 2001L, 1L))
-                .addLinkedEvents(new DetectionLinkedEvent(stroom, 2002L, 2L))
-                .build();
+                .withEffectiveExecutionTime(effectiveExecutionTime);
+
+        if (includeValues) {
+            builder
+                    .addValue("name-1", "value-A")
+                    .addValue("name-2", "value-B")
+                    .addValue("name-3", "value-C")
+                    .addValue("name-4", null);
+        }
+        if (includeLinkedEvents) {
+            builder
+                    .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 1L))
+                    .addLinkedEvents(new DetectionLinkedEvent(stroom, 1001L, 2L))
+                    .addLinkedEvents(new DetectionLinkedEvent(stroom, 2001L, 1L))
+                    .addLinkedEvents(new DetectionLinkedEvent(stroom, 2002L, 2L));
+        }
+        return builder.build();
     }
 }

--- a/stroom-config/stroom-config-app/src/test/resources/stroom/config/app/expected.yaml
+++ b/stroom-config/stroom-config-app/src/test/resources/stroom/config/app/expected.yaml
@@ -1110,7 +1110,7 @@ appConfig:
         \  <ul>\n    {% for linkedEvent in linkedEvents -%}\n      <li>Environment:\
         \ {{ linkedEvent.stroom | escape }}, Stream ID: {{ linkedEvent.streamId |\
         \ escape }}, Event ID: {{ linkedEvent.eventId | escape }}</li>\n    {% endfor\
-        \ %}\n  </ul>\n  {% endif %}\n</body>\n{% endif %}\n"
+        \ %}\n  </ul>\n  {% endif %}\n</body>\n"
       defaultSubjectTemplate: "Detector '{{ detectorName | escape }}' Alert"
     defaultMaxResults: "1000000,100,10,1"
     helpSubPathDocumentation: "/user-guide/content/documentation/"

--- a/stroom-core-shared/src/main/java/stroom/ui/config/shared/AnalyticUiDefaultConfig.java
+++ b/stroom-core-shared/src/main/java/stroom/ui/config/shared/AnalyticUiDefaultConfig.java
@@ -32,15 +32,15 @@ public class AnalyticUiDefaultConfig extends AbstractConfig implements IsStroomC
     @JsonProperty
     @JsonPropertyDescription(
             "The default email subject template to use for analytic rule emails. The template uses " +
-            "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
-            "with an initial subject template value.")
+                    "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
+                    "with an initial subject template value.")
     private final String defaultSubjectTemplate;
 
     @JsonProperty
     @JsonPropertyDescription(
             "The default email body template to use for analytic rule emails. The template uses " +
-            "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
-            "with an initial body template value.")
+                    "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
+                    "with an initial body template value.")
     private final String defaultBodyTemplate;
 
     @SuppressWarnings("checkstyle:LineLength")
@@ -52,34 +52,33 @@ public class AnalyticUiDefaultConfig extends AbstractConfig implements IsStroomC
         //noinspection TextBlockMigration // GWT no likey textblock, grrr!
         defaultBodyTemplate =
                 "<!DOCTYPE html>\n" +
-                "<html lang=\"en\">\n" +
-                "<meta charset=\"UTF-8\" />\n" +
-                "<title>Detector '{{ detectorName | escape }}' Alert</title>\n" +
-                "<body>\n" +
-                "  <p>Detector <em>{{ detectorName | escape }}</em> " +
-                "{{ detectorVersion | escape }} fired at {{ detectTime | escape }}</p>\n" +
-                "\n" +
-                "  {%- if (values | length) > 0 -%}\n" +
-                "  <p>Detail: {{ headline | escape }}</p>\n" +
-                "  <ul>\n" +
-                "    {% for key, val in values | dictsort -%}\n" +
-                "      <li><strong>{{ key | escape }}</strong>: {{ val | escape }}</li>\n" +
-                "    {% endfor %}\n" +
-                "  </ul>\n" +
-                "  {% endif -%}\n" +
-                "\n" +
-                "  {%- if (linkedEvents | length) > 0 -%}\n" +
-                "  <p>Linked Events:</p>\n" +
-                "  <ul>\n" +
-                "    {% for linkedEvent in linkedEvents -%}\n" +
-                "      <li>Environment: {{ linkedEvent.stroom | escape }}, " +
-                "Stream ID: {{ linkedEvent.streamId | escape }}, " +
-                "Event ID: {{ linkedEvent.eventId | escape }}</li>\n" +
-                "    {% endfor %}\n" +
-                "  </ul>\n" +
-                "  {% endif %}\n" +
-                "</body>\n" +
-                "{% endif %}\n";
+                        "<html lang=\"en\">\n" +
+                        "<meta charset=\"UTF-8\" />\n" +
+                        "<title>Detector '{{ detectorName | escape }}' Alert</title>\n" +
+                        "<body>\n" +
+                        "  <p>Detector <em>{{ detectorName | escape }}</em> " +
+                        "{{ detectorVersion | escape }} fired at {{ detectTime | escape }}</p>\n" +
+                        "\n" +
+                        "  {%- if (values | length) > 0 -%}\n" +
+                        "  <p>Detail: {{ headline | escape }}</p>\n" +
+                        "  <ul>\n" +
+                        "    {% for key, val in values | dictsort -%}\n" +
+                        "      <li><strong>{{ key | escape }}</strong>: {{ val | escape }}</li>\n" +
+                        "    {% endfor %}\n" +
+                        "  </ul>\n" +
+                        "  {% endif -%}\n" +
+                        "\n" +
+                        "  {%- if (linkedEvents | length) > 0 -%}\n" +
+                        "  <p>Linked Events:</p>\n" +
+                        "  <ul>\n" +
+                        "    {% for linkedEvent in linkedEvents -%}\n" +
+                        "      <li>Environment: {{ linkedEvent.stroom | escape }}, " +
+                        "Stream ID: {{ linkedEvent.streamId | escape }}, " +
+                        "Event ID: {{ linkedEvent.eventId | escape }}</li>\n" +
+                        "    {% endfor %}\n" +
+                        "  </ul>\n" +
+                        "  {% endif %}\n" +
+                        "</body>\n";
     }
 
     @JsonCreator
@@ -119,12 +118,12 @@ public class AnalyticUiDefaultConfig extends AbstractConfig implements IsStroomC
     @Override
     public String toString() {
         return "AnalyticUiDefaultConfig{" +
-               "defaultNode='" + defaultNode + '\'' +
-               ", defaultErrorFeed=" + defaultErrorFeed +
-               ", defaultDestinationFeed=" + defaultDestinationFeed +
-               ", defaultSubjectTemplate='" + defaultSubjectTemplate + '\'' +
-               ", defaultBodyTemplate='" + defaultBodyTemplate + '\'' +
-               '}';
+                "defaultNode='" + defaultNode + '\'' +
+                ", defaultErrorFeed=" + defaultErrorFeed +
+                ", defaultDestinationFeed=" + defaultDestinationFeed +
+                ", defaultSubjectTemplate='" + defaultSubjectTemplate + '\'' +
+                ", defaultBodyTemplate='" + defaultBodyTemplate + '\'' +
+                '}';
     }
 
     @Override
@@ -137,10 +136,10 @@ public class AnalyticUiDefaultConfig extends AbstractConfig implements IsStroomC
         }
         final AnalyticUiDefaultConfig that = (AnalyticUiDefaultConfig) object;
         return Objects.equals(defaultNode, that.defaultNode)
-               && Objects.equals(defaultErrorFeed, that.defaultErrorFeed)
-               && Objects.equals(defaultDestinationFeed, that.defaultDestinationFeed)
-               && Objects.equals(defaultSubjectTemplate, that.defaultSubjectTemplate)
-               && Objects.equals(defaultBodyTemplate, that.defaultBodyTemplate);
+                && Objects.equals(defaultErrorFeed, that.defaultErrorFeed)
+                && Objects.equals(defaultDestinationFeed, that.defaultDestinationFeed)
+                && Objects.equals(defaultSubjectTemplate, that.defaultSubjectTemplate)
+                && Objects.equals(defaultBodyTemplate, that.defaultBodyTemplate);
     }
 
     @Override

--- a/stroom-core-shared/src/main/java/stroom/ui/config/shared/AnalyticUiDefaultConfig.java
+++ b/stroom-core-shared/src/main/java/stroom/ui/config/shared/AnalyticUiDefaultConfig.java
@@ -32,15 +32,15 @@ public class AnalyticUiDefaultConfig extends AbstractConfig implements IsStroomC
     @JsonProperty
     @JsonPropertyDescription(
             "The default email subject template to use for analytic rule emails. The template uses " +
-                    "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
-                    "with an initial subject template value.")
+                    "a sub-set of the Jinja templating language. If this property is not set, the user " +
+                    "will not be presented with an initial subject template value.")
     private final String defaultSubjectTemplate;
 
     @JsonProperty
     @JsonPropertyDescription(
             "The default email body template to use for analytic rule emails. The template uses " +
-                    "a sub-set of the Jinja templating language. If this property is not set, the user will not be presented " +
-                    "with an initial body template value.")
+                    "a sub-set of the Jinja templating language. If this property is not set, the user will " +
+                    "not be presented with an initial body template value.")
     private final String defaultBodyTemplate;
 
     @SuppressWarnings("checkstyle:LineLength")

--- a/unreleased_changes/20240919_131039_309__4463.md
+++ b/unreleased_changes/20240919_131039_309__4463.md
@@ -1,0 +1,24 @@
+* Issue **#4463** : Fix NPE with analytic rule email templating.
+
+
+```sh
+# ********************************************************************************
+# Issue title: v7.4 Email NPE when Alerting a Streaming Analytic
+# Issue link:  https://github.com/gchq/stroom/issues/4463
+# ********************************************************************************
+
+# ONLY the top line will be included as a change entry in the CHANGELOG.
+# The entry should be in GitHub flavour markdown and should be written on a SINGLE
+# line with no hard breaks. You can have multiple change files for a single GitHub issue.
+# The  entry should be written in the imperative mood, i.e. 'Fix nasty bug' rather than
+# 'Fixed nasty bug'.
+#
+# Examples of acceptable entries are:
+#
+#
+# * Issue **123** : Fix bug with an associated GitHub issue in this repository
+#
+# * Issue **namespace/other-repo#456** : Fix bug with an associated GitHub issue in another repository
+#
+# * Fix bug with no associated GitHub issue.
+```

